pick.lic

```diff
@@ -2,523 +2,665 @@
+      remove_hindering_gear
+      crack_boxes
+      stop_buffs
+      refill_ring
+    end
   end
 
-  def check_pet_boxes(source)
-    return unless source
+  def open_containers
+    @sources.each do |source_container|
+      DRCI.open_container?(source_container)
+    end
+    DRCI.open_container?(@skipped_box_storage) if @skipped_box_storage
+    DRCI.open_container?(@component_container) if @component_container
+  end
 
-    boxes = get_boxes(source)
-    return if boxes.empty?
+  def check_for_boxes
+    @boxes_by_bag = {}
+    @sources.each do |source_container|
+      @boxes_by_bag[source_container] = DRC.get_boxes(source_container)
+    end
+    if @boxes_by_bag.values.flatten.empty?
+      echo "No boxes to pop"
+      exit
+    end
+  end
 
-    @harvest_traps = false
+  def remove_hindering_gear
+    @equipment_manager.empty_hands
 
-    boxes.each do |box|
-      if DRSkill.getxp('Locksmithing') >= 30
-        echo '***STOPPING DUE TO MINDLOCK***'
-        break
-      end
+    @removed_items = @equipment_manager.remove_gear_by(&:hinders_lockpicking)
 
-      bput("get my #{box} from my #{source}", 'You get a .* from inside ')
-      unless disarm?(box)
-        bput("put my #{box} in my #{source}", 'You put your .* in ') if right_hand
-        next
-      end
+    if DRC.left_hand || DRC.right_hand
+      DRC.message('***ITEMS ARE STILL IN HANDS, EXITING***')
+      @equipment_manager.wear_items(@removed_items)
+      exit
+    end
+  end
 
-      find_lockpick unless @settings.use_lockpick_ring
+  def setup_flags
+    Flags.add('disarm-shift', 'something to shift')
+    Flags.add('disarm-trap-type', *@all_trap_messages.values)
+    Flags.add('more-traps', 'not fully disarmed', 'not yet fully disarmed', 'still has more to torment you with')
+    Flags.add('more-locks', 'You discover another lock protecting')
+    Flags.add('glance-no-traps', 'It looks like there are no traps left on')
+    Flags.add('glance-no-locks', 'It looks like there are no locks left on')
+  end
 
-      loop do
-        if DRSkill.getxp('Locksmithing') >= 30
-          bput("put my #{box} in my #{source}", 'You put your .* in ') if right_hand
-          fput('stow left') unless @settings.use_lockpick_ring
-          break
-        end
+  def crack_boxes
+    @boxes_by_bag.each do |source, boxes|
+      boxes.each do |box_noun|
+        break if stop_picking?
 
-        case bput("pick my #{box} blind", 'not even locked', 'Roundtime', 'Find a more appropriate tool')
-        when 'not even locked'
-          waitrt?
-          fput('stow left') unless @settings.use_lockpick_ring
-          loot(box)
-          dismantle(box)
-          break
-        when 'Find a more appropriate tool'
-          bput("put my #{box} in my #{source}", 'You put your .* in ') if right_hand
-          fput('stow left') unless @settings.use_lockpick_ring
-          break
-        end
+        do_buffs
+        DRC.bput('sit', 'You sit', 'You are already sitting') unless sitting?
```

Comment:
Adding more match strings from when I worked on sitting to perform barb meditations in `DRCA.activate_barb_buff?`
```suggestion
        DRC.bput('sit', 'You sit', 'You are already sitting', 'You rise', 'While swimming?') unless sitting?
```

Reply:
Feels like maybe we're missing some utilities somewhere for posture (?) control. "what posture am I in?" and "sit" "kneel" etc. as common utils so we don't have these match strings everywhere.